aiida_vasp/parsers/file_parsers/vasprun.py

```diff
@@ -651,6 +679,58 @@ def fermi_level(self):
 
         return self._xml.get_fermi_level()
 
+    @property
+    def band_properties(self):
```

Comment:
This we already have in `bands`. And the bandgap etc. can be calculated using AiiDA routines. Indeed, that lacks VMB/CBM and I also want this. I think we instead should make a PR to core regarding that and continue to use `bands`. Or is there any reason why we can not use `bands` as it is?

Reply:
hmm, I see. The point here is to add some information to give a hint about the system, e.g. whether it is a conducting or insulating. Typically when one does a regular single point/relaxation using moderate grid density there is not much point for parsing and storing the full `BandsData`. 

We should include CBM/VBM in `bands` as well.

Reply:
We should never use the energies from the relaxation run directly. Always run again without relaxation. But I guess you already know that, so I am just noting this here for reference. Yes, what I think would be nice to have is to not store the `BandsData`. So I would maybe instead suggest that we use the `BandsData` framework (so that we can utilize and contribute common functionality) and then not store it if we want to just get say VBM/CBM etc. I think we should be able to do this quite easily in the parser. E.g. create a `BandsData`, use the band gap utils (and making a PR for VBM/CBM) etc. and then only store that and discard `BandsData`. Does this sound okey?

Reply:
I cannot see a way to create a `BandsData`, parse it for the CBM/VBM without having it stored, unless it is sensible to import `NodeComposer` into the file parser modules or copy and paste `_compose_array_bands` in? 
Also there are additional overheads in creating the `BandsData` itself.

Reply:
The `eigenvalues` and `occupations` are already in the way that can be directly utilised by `BandsData`. I think there isn't much point of writing another wrapper just for the sake of it.

Reply:
We do not use the `add_bands`, but use  a clean `BandsData` instance in the routine to calculate the bandgap etc. This would not be stored if we do not set it on a node or call `store`. In this way `aiida_core` users would benefit from any improvements we add  to the existing routines and they would be calculated in the same way. `eigenvalues` and `occupations` is fine. So we should maybe just refactor the code for `bands` so that we can use a utility function to create a `BandsData`. Then in `bands` we set it on the node and then it is stored, while for the properties we use it to call the routines in AiiDA for calculating the bandgap (e.g. `mybandsdata.find_bandgap()`) etc.

Reply:
I am not very sure if I understand your discussion correctly, but does `BandsData` have the method to calculate bandgap?

I have an impression that `BandsData` aims to store band structure along high-symmetry paths. But VBM and CBM may not be on the lines. So if we want to reuse the feature in aiida-core, we should make `find_bandgap` in `bands.py` to accept general (energies, occupations) pair, or define a class inherited from `KpointsData` that mimics
```
_, stored_occupations = bandsdata.get_bands(also_occupations=True)
```
This way, we can get bandgap, but I don't see `find_bandgap` returns VBM and CBM.

Reply:
@espenfl My confusion is about where should the function for getting the required `band_properties` from `BandsData` be placed, shoud it be in the `NodeComposer._compose_dict` and `_compose_array_bands`? 

The point of adding these quantities is just add a bit more metadata for regular calculations. They are probably not physically correct for calculations with grids, but they are still well-defined for the calcualtions themselves. This way the user can quickly know if the material is conducting or insulting.